docs/templates.md

```diff
@@ -129,6 +130,20 @@ The following methods are defined.
 * `.short([len: Integer]) -> String`
 * `.shortest([min_len: Integer]) -> ShortestIdPrefix`: Shortest unique prefix.
 
+### CryptographicSignature type
+
+The following methods are defined.
+
+* `.status() -> String`: The signature's status ("good", "bad", "unknown", "invalid").
+* `.key() -> String`: The signature's key id representation (for GPG, this is the key fingerprint).
+* `.display() -> String`: The signature's display string (for GPG this is the formatted primary user ID.
+
+!!! warning
+
+    Calling any of `.status()`, `.key()`, or `.display()` is slow, as it incurs
+    the performance cost of verifying the signature. Though consecutive calls
+    will be faster, because the backend caches the verification result.
```

Comment:
(actually looking through the code top-to-bottom finally)

imo you should mention that just checking for signature presence via the boolean coersion (`if(commit.signature(), "commit has a signature", "commit is unsigned")`) is very fast and cheap, as opposed to the verification part

Also, `[..] verifying the signature. Though [..]` -> `[..] verifying the signature (for example, shelling out to gpg or ssh-keygen). Though [..]`, just giving an example of why it is slow.

fyi creating a single commit with a gpg signature made through yubikey takes ~200ms for me, a bit annoying :)

fyi² it is (by design) possible to implement another gpg backend (for example) that'll be faster by using some rust gpg implementation or something instead of the lazy way of just calling the gpg binary.

Reply:
Let me know if I am overdoing it with [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/). I *love* admonitions. 😅